src/server/i18n/index.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { i18n, i18nLoader } from '@kbn/i18n';
+
+export async function i18nMixin(kbnServer, server, config) {
```

Comment:
__nit/question:__ how do you feel about collapsing that into something like this (haven't tested, so the code may not work in this exact form :)):
```js
export async function i18nMixin(kbnServer, server, config) {
  i18nLoader.registerTranslationFiles(kbnServer.uiExports.translationPaths || []);

  const translations = Object.freeze(await i18nLoader.getTranslationsByLocale(config.get('i18n.locale')));
  i18n.init(translations);

  server.decorate('server', 'getUiTranslations', () => translations);
}
```